cel/program.go

```diff
@@ -207,6 +207,30 @@ func newProgram(e *Env, ast *Ast, opts []ProgramOption) (Program, error) {
 	if len(p.regexOptimizations) > 0 {
 		decorators = append(decorators, interpreter.CompileRegexConstants(p.regexOptimizations...))
 	}
+	// Enable compile-time checking of syntax/cardinality for string.format calls.
+	if p.evalOpts&OptCheckStringFormat == OptCheckStringFormat {
+		enableCheckArgTypes := ast.IsChecked()
```

Comment:
I'm not quite sure I understand here - how would we go about doing that? I don't think there's any function on `ref.Val` that would support doing something like that.

Reply:
A `typeVerifier` which returns trivially valid for expressions that haven't been type-checked (`!ast.IsChecked()`) would allow you to remove checks for whether to type-check args later on.

Reply:
Oh, makes sense, so something like:

```golang
func typeVerifier(...) bool, error {
    if !ast.IsChecked() {
        return true, nil
    }
     // tests in case we're dealing with a type-checked expression  
}
```

I think the tricky part would be doing this and lifting `typeVerifier` out as you mentioned, since we need AST access for that - I don't have a good answer for that either at the moment unfortunately, but I'll see if I can come up with one.

Reply:
More like you could either skip creating this decorator at all if the ast is not checked, or you could create an `isValidType` function that just returns `true, nil`

```
isValidType := func(int64, ...*types.TypeValue) (bool, error) {
   return true, nil
}
if ast.IsChecked() {
  isValidType = ...
}
```

Reply:
(marking as resolved, `checkEnableArgTypes` is now removed in favor of changing the body of `isValidType`)